docs/design-docs/02-custom-root-reconciler-clusterrole.md

```diff
@@ -1,15 +1,21 @@
 # Custom `ClusterRole` assignments for `RootSync`s
 
-* Author(s): @tomasaschan
-* Approver: @sdowell, @janetkuo
+* Author(s): @tomasaschan, @sdowell
+* Approver: @mortent, @karlkfi
 * Status: approved
 
 ## Summary
 
 When creating a reconciler deployment for a `RootSync`, Config Sync currently [creates a
 `ClusterRoleBinding`] for the generated `ServiceAccount` granting it `cluster-admin`.
 
-This proposal introduces a mechanism to customize this behavior.
+On the other hand, when creating a reconciler deployment for a `RepoSync`, Config Sync
```

Comment:
Is this right? I thought this was just how the e2e tests work. In normal use, the user has to create a role & rolebinding, right?

Reply:
What you're thinking of is something the e2e tests used to do - they always created a RoleBinding which granted cluster-admin so that RepoSyncs could manage any resource in the namespace. This has since been changed, and each e2e test needs to actually declare which permissions are granted to the RepoSync reconciler. This enables more realistic testing of RepoSync reconcilers and their typical RoleBindings.

The `configsync.gke.io:ns-reconciler` RoleBinding here is separate, and this RoleBinding is always created. The RoleBinding binds to the ClusterRole [here](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/manifests/ns-reconciler-cluster-role.yaml), which just grants basic permissions such as RepoSync status updates etc. My understanding is that this is for convenience, as this is the bare minimum that is necessary for a RepoSync reconciler to run and would just add burden for the user to create.

I realized that the similar concept applies to RootSync if we strip the cluster-admin binding, so I added a similar base ClusterRole for RootSync that will be automatically created.

Reply:
Ahh. ok. That wasn't clear from this doc.

I think perhaps it could be clarified with a before and after table or list or something.